### PR TITLE
13976 segments not deleted

### PR DESCRIPTION
**Pull request by flamisz**
### Description:

fix for #13976 

### Review

* [ ] Functional review done
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by flamisz** [comment]
Hi all, this PR is not "finished" yet, I'd like to add an extra test, any suggestion is appreciated.
The test that is already included is testing if the `transferAllUserSegmentsToSuperUser` is working properly, but one extra test is needed to make sure we actually call this function when a user is deleted.

It's happening in `plugins/CoreAdminHome/CoreAdminHome.php` on a `UsersManager.deleteUser` event. What's the best way to test this?

Thanks

**Comment by tsteur** [comment]
Hi @flamisz if you wanted to test that it's actually called you could basically first check there are segments for that user like `$this->assertNotEmpty($segments->getsegments...)` delete that user (eg using UsersManager API) and check that no segments are defined for that user anymore like `$this->assertEmpty($segment->...)`. It wouldn't need a detailed test or so since the method itself is already tested.

File: plugins/CoreAdminHome/CoreAdminHome.php
Comment (tsteur) [inline plugins/CoreAdminHome/CoreAdminHome.php]: btw usually we would call this method in `Piwik\Plugins\SegmentEditor\SegmentEditor` class by listening to the `UsersManager.deleteUser` event see https://developer.matomo.org/guides/events . There we would then call the method to delete all segments for the deleted user. Basically the same way it's done in this class. It's not needed to do it that way but usually best practice to contain all the logic that belongs to the plugin in the actual plugin.

Comment (flamisz) [reply plugins/CoreAdminHome/CoreAdminHome.php]: I like that, more clean solution. I will modify it.

**Comment by diosmosis** [comment]
Code looks good, but it appears to be causing another integration test to fail.

**Comment by flamisz** [comment]
> Code looks good, but it appears to be causing another integration test to fail.

@diosmosis the test had errors in the `tearDown()`. I fixed those. it should be fine, will see soon :)

(of course the error was caused by my pr)

**Review by diosmosis — APPROVED** [review]
looks good, will merge tomorrow after a bit of time's passed since the 4.2.0 release

File: plugins/SegmentEditor/SegmentEditor.php
Comment (diosmosis) [inline plugins/SegmentEditor/SegmentEditor.php]: Sorry, noticed one other issue: is there a reason this method is `static`? If not let me know and I'll change it before merging.

Comment (flamisz) [reply plugins/SegmentEditor/SegmentEditor.php]: definitely NOT, I'm not sure why it is static, sorry. thanks for changing it @diosmosis

**Pull request merged**
